src/polar/kernels.jl

```diff
@@ -244,6 +307,52 @@ function transfer!(polar::PolarForm, buffer::PolarNetworkState, u)
     wait(ev)
 end
 
+KA.@kernel function adj_transfer_kernel!(
+    adj_vmag, adj_vang, adj_pinj, adj_qinj, adj_u, pv, pq, ref, adj_pload, adj_qload
```

Comment:
I guess here `adj_pload` is the adjoint of the power generation `pg` (`adj_pg`), as the load is constant?

Reply:
I am not sure. In that case we could remove that argument. I am not yet sure how to handle the `adjoint!` signatures in general. What we also have to clarify is weither the adjoint routines return a primal value. I would not recommend that.